packages/grid/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -37,174 +45,238 @@ export const GridRootStyles = styled('div', {
-    { [`& .${gridClasses.treeDataGroupingCellToggle}`]: styles.treeDataGroupingCellToggle },
-    { [`& .${gridClasses.detailPanelToggleCell}`]: styles.detailPanelToggleCell },
+      [`& .${c['columnHeader--withRightBorder']}`]: styles['columnHeader--withRightBorder'],
+    },
+    { [`& .${c.columnHeader}`]: styles.columnHeader },
+    { [`& .${c.headerFilterRow}`]: styles.headerFilterRow },
+    { [`& .${c.columnHeaderCheckbox}`]: styles.columnHeaderCheckbox },
+    { [`& .${c.columnHeaderDraggableContainer}`]: styles.columnHeaderDraggableContainer },
+    { [`& .${c.columnHeaderTitleContainer}`]: styles.columnHeaderTitleContainer },
+    { [`& .${c['columnSeparator--resizable']}`]: styles['columnSeparator--resizable'] },
+    { [`& .${c['columnSeparator--resizing']}`]: styles['columnSeparator--resizing'] },
+    { [`& .${c.columnSeparator}`]: styles.columnSeparator },
+    { [`& .${c.filterIcon}`]: styles.filterIcon },
+    { [`& .${c.iconSeparator}`]: styles.iconSeparator },
+    { [`& .${c.menuIcon}`]: styles.menuIcon },
+    { [`& .${c.menuIconButton}`]: styles.menuIconButton },
+    { [`& .${c.menuOpen}`]: styles.menuOpen },
+    { [`& .${c.menuList}`]: styles.menuList },
+    { [`& .${c['row--editable']}`]: styles['row--editable'] },
+    { [`& .${c['row--editing']}`]: styles['row--editing'] },
+    { [`& .${c['row--dragging']}`]: styles['row--dragging'] },
+    { [`& .${c.row}`]: styles.row },
+    { [`& .${c.rowReorderCellPlaceholder}`]: styles.rowReorderCellPlaceholder },
+    { [`& .${c.rowReorderCell}`]: styles.rowReorderCell },
+    { [`& .${c['rowReorderCell--draggable']}`]: styles['rowReorderCell--draggable'] },
+    { [`& .${c.sortIcon}`]: styles.sortIcon },
+    { [`& .${c.withBorderColor}`]: styles.withBorderColor },
+    { [`& .${c.treeDataGroupingCell}`]: styles.treeDataGroupingCell },
+    { [`& .${c.treeDataGroupingCellToggle}`]: styles.treeDataGroupingCellToggle },
+    { [`& .${c.detailPanelToggleCell}`]: styles.detailPanelToggleCell },
     {
-      [`& .${gridClasses['detailPanelToggleCell--expanded']}`]:
-        styles['detailPanelToggleCell--expanded'],
+      [`& .${c['detailPanelToggleCell--expanded']}`]: styles['detailPanelToggleCell--expanded'],
     },
     styles.root,
   ],
-})<{ ownerState: OwnerState }>(({ theme }) => {
-  const borderColor = getBorderColor(theme);
-  const radius = theme.shape.borderRadius;
+})<{ ownerState: OwnerState }>(({ theme: t }) => {
+  const borderColor = getBorderColor(t);
+  const radius = t.shape.borderRadius;
+
+  const containerBackground = t.vars
+    ? t.vars.palette.background.defaultChannel
+    : t.palette.background.default;
+
+  const pinnedBackground =
+    t.palette.mode === 'dark'
+      ? lighten(
+          t.vars ? t.vars.palette.background.defaultChannel : t.palette.background.default,
+          0.1,
+        )
+      : darken(
+          t.vars ? t.vars.palette.background.defaultChannel : t.palette.background.default,
+          0.05,
+        );
```

Comment:
The header background, which now needs a solid background color, is from the palette's background color. The pinned cells' background is derived from the same value, but lightened or darkened.

I'm not sure what's the best way to make the pinned background customizable with this setup, suggestions welcome.

Reply:
What are the reasons for a different bgcolor for pinned columns?
I would favor using the same bgcolor for pinned and non-pinned columns. The shadow that indicates the pinned columns should be enough.

Reply:
I've used the docs as my reference to restore the design after I do changes:
![image](https://github.com/mui/mui-x/assets/1423607/f41fbd89-b386-477a-bb5e-73a66daf3da3)

But I just noticed that light theme doesn't differentiate. I never use light theme.

Reply:
This is out of the scope of this PR, but this reminds me of the discussion we had in https://github.com/mui/mui-x/issues/5979#issuecomment-1235812999
I feel like using the same bg color for pinned columns (in both light and dark modes) looks better, as long as we use a thicker border or a shadow to mark the boundaries between pinned/unpinned columns - see https://www.figma.com/proto/W6TItYkiGJZjL5zLnfeSjy/Data-grid?node-id=511-25798&scaling=min-zoom&page-id=0%3A1&starting-point-node-id=501%3A31128

Reply:
Sounds good for using the same color for pinned rows. The shadow styles are broken right now anyway, I need to work on that. Do we have a preference?

Reply:
For me, the ideal behavior is:
- same bg color for pinned and non-pinned columns
- conditional shadow on pinned columns: see the demo in https://github.com/mui/mui-x/issues/7152#issuecomment-1344981248

Would you like to work on this after merging this PR?
There's an issue for this enhancement already: https://github.com/mui/mui-x/issues/10001